generate_nb_vocab_file.py

```diff
@@ -0,0 +1,54 @@
+import inspect
+from pathlib import Path
+
+import pydantic
+
+from bagel import mappings, models
+
+EXCLUDE_CLASSES = ["Bagel", "BaseModel"]
+PREAMBLE = "\n".join(
+    [
+        "@prefix nb: <http://neurobagel.org/vocab/>.",
+        "@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#>.",
+    ]
+)
+OUTPUT_PATH = Path.cwd() / "nb_vocab.ttl"
+
+
+def generate_vocab_string_from_models():
+    all_classes = {
+        name: obj
+        for name, obj in inspect.getmembers(models)
+        if inspect.isclass(obj)
+        and issubclass(obj, pydantic.BaseModel)
+        and name not in EXCLUDE_CLASSES
+    }
+    nb_vocab = ""
+
+    for count, (name, obj) in enumerate(sorted(all_classes.items())):
+        class_entry = f"{mappings.NB.pf}:{name}\n" "    a rdfs:Class"
+
+        is_subclass = False
+        for other_name, other_obj in all_classes.items():
+            if name != other_name and issubclass(obj, other_obj):
+                class_entry += (
+                    f";\n    rdfs:subClassOf {mappings.NB.pf}:{other_name}."
+                )
+                is_subclass = True
+                break
+
+        if not is_subclass:
+            class_entry += "."
```

Comment:
This should be easier I think. Could you loop [over `obj.__bases__` ](https://docs.python.org/3/reference/datamodel.html#type.__bases__)to see if there are any that are not a BaseModel? I guess that would give you the parents, not the children. But maybe you could pre-sort the models who inherit from classes other than BaseModel and then deal with those models separately.

Reply:
oh wait, I think finding the parents of `obj` *would* be enough here, no?